src/Common/src/System/Drawing/ColorTranslator.cs

```diff
@@ -391,34 +402,36 @@ public static string ToHtml(Color c)
 
         private static void InitializeHtmlSysColorTable()
         {
-            s_htmlSysColorTable = new Hashtable(26);
-            s_htmlSysColorTable["activeborder"] = ColorUtil.FromKnownColor(KnownColor.ActiveBorder);
-            s_htmlSysColorTable["activecaption"] = ColorUtil.FromKnownColor(KnownColor.ActiveCaption);
-            s_htmlSysColorTable["appworkspace"] = ColorUtil.FromKnownColor(KnownColor.AppWorkspace);
-            s_htmlSysColorTable["background"] = ColorUtil.FromKnownColor(KnownColor.Desktop);
-            s_htmlSysColorTable["buttonface"] = ColorUtil.FromKnownColor(KnownColor.Control);
-            s_htmlSysColorTable["buttonhighlight"] = ColorUtil.FromKnownColor(KnownColor.ControlLightLight);
-            s_htmlSysColorTable["buttonshadow"] = ColorUtil.FromKnownColor(KnownColor.ControlDark);
-            s_htmlSysColorTable["buttontext"] = ColorUtil.FromKnownColor(KnownColor.ControlText);
-            s_htmlSysColorTable["captiontext"] = ColorUtil.FromKnownColor(KnownColor.ActiveCaptionText);
-            s_htmlSysColorTable["graytext"] = ColorUtil.FromKnownColor(KnownColor.GrayText);
-            s_htmlSysColorTable["highlight"] = ColorUtil.FromKnownColor(KnownColor.Highlight);
-            s_htmlSysColorTable["highlighttext"] = ColorUtil.FromKnownColor(KnownColor.HighlightText);
-            s_htmlSysColorTable["inactiveborder"] = ColorUtil.FromKnownColor(KnownColor.InactiveBorder);
-            s_htmlSysColorTable["inactivecaption"] = ColorUtil.FromKnownColor(KnownColor.InactiveCaption);
-            s_htmlSysColorTable["inactivecaptiontext"] = ColorUtil.FromKnownColor(KnownColor.InactiveCaptionText);
-            s_htmlSysColorTable["infobackground"] = ColorUtil.FromKnownColor(KnownColor.Info);
-            s_htmlSysColorTable["infotext"] = ColorUtil.FromKnownColor(KnownColor.InfoText);
-            s_htmlSysColorTable["menu"] = ColorUtil.FromKnownColor(KnownColor.Menu);
-            s_htmlSysColorTable["menutext"] = ColorUtil.FromKnownColor(KnownColor.MenuText);
-            s_htmlSysColorTable["scrollbar"] = ColorUtil.FromKnownColor(KnownColor.ScrollBar);
-            s_htmlSysColorTable["threeddarkshadow"] = ColorUtil.FromKnownColor(KnownColor.ControlDarkDark);
-            s_htmlSysColorTable["threedface"] = ColorUtil.FromKnownColor(KnownColor.Control);
-            s_htmlSysColorTable["threedhighlight"] = ColorUtil.FromKnownColor(KnownColor.ControlLight);
-            s_htmlSysColorTable["threedlightshadow"] = ColorUtil.FromKnownColor(KnownColor.ControlLightLight);
-            s_htmlSysColorTable["window"] = ColorUtil.FromKnownColor(KnownColor.Window);
-            s_htmlSysColorTable["windowframe"] = ColorUtil.FromKnownColor(KnownColor.WindowFrame);
-            s_htmlSysColorTable["windowtext"] = ColorUtil.FromKnownColor(KnownColor.WindowText);
+            s_htmlSysColorTable = new Hashtable(26)
```

Comment:
There are 27 entries here, but the capacity is being set to 26?

Reply:
Missed that existing bit- we shouldn't have to explicitly set it anyway, right?

Reply:
> we shouldn't have to explicitly set it anyway, right?

You don't _have_ to, but doing so allows it to be presized, which then avoids additional array allocations internally as the table grows to accomodate the additional incrementally added items, e.g.
```C#
using System;
using System.Collections;

class Program
{
    static void Main()
    {
        foreach (int size in new[] { 27, 26, 0 })
        {
            long mem = GC.GetAllocatedBytesForCurrentThread();
            var ht = size == 0 ? new Hashtable() : new Hashtable(size);
            for (int i = 0; i < 27; i++) ht.Add(i, "");
            mem = GC.GetAllocatedBytesForCurrentThread() - mem;
            Console.WriteLine($"{size.ToString().PadRight(2)}: {mem} byes");
        }
    }
}
```
outputs:
```
27: 3792 byes
26: 3792 byes
0 : 4512 byes
```

Reply:
I see. I presumed (incorrectly) that there would be smarts for this in the compiler.

Reply:
I'm going to change this to a dictionary. Cuts the allocated size down 1440 bytes and removes the boxing.

Reply:
Surprisingly a collection initializer does not attempt to presize. Eg., `var d = new Dictionary<int, int> { { 1, 2 }, { 2, 3 }, { 3, 4 } };` produces `newobj     instance void class [System.Collections]System.Collections.Generic.Dictionary`2<int32,int32>::.ctor()`. I would have thought the compiler could assume that "3" was better than the default here (although it is arguable)

Reply:
> Surprisingly a collection initializer does not attempt to presize

How would the collection initializer know what constructor to use?  Look for one that takes an `int` and hope for the best?